### PR TITLE
Complete Friends App

### PR DESCRIPTION
**Pull request by bugagashinka**
### 14. Friends App

[Demo](https://bugagashinka.github.io/friends-app/) |
[Code base](https://github.com/bugagashinka/friends-app)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/bugagashinka/friends-app/js/components/DataModel.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        fullName2 = `${n2.first} ${n2.last}`;
+      if (fullName1 < fullName2) return asc ? -1 : 1;
+      if (fullName1 > fullName2) return asc ? 1 : -1;
+    };
+  };
+
+  const resetOrderFilter = ({ age = false, name = false, asc = true } = {}) => {
+    isOrderedBy.age = age;
+    isOrderedBy.name = name;
+    isOrderedBy.asc = asc;
```
Comment (yevhenorlov) [inline submissions/bugagashinka/friends-app/js/components/DataModel.js]: es6 syntax trick, you can do it via spread operator in one line:
```js
isOrderedBy = {...isOrderedBy, age, name, asc}
```

File: submissions/bugagashinka/friends-app/index.html [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Friends App</title>
+    <link rel="stylesheet" href="styles/main.css" />
+    <link
+      rel="stylesheet"
+      href="https://use.fontawesome.com/releases/v5.6.3/css/all.css"
+    />
+  </head>
+  <body>
+    <head></head>
```
[the remark this answers is not in the document]
Comment (bugagashinka) [reply submissions/bugagashinka/friends-app/index.html]: Yes, thank. It's should be header

File: submissions/bugagashinka/friends-app/index.html [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    <main class="wrapper">
+      <section id="filter">
+        <h2>Filter:</h2>
+        <hr />
```
[the remark this answers is not in the document]
Comment (bugagashinka) [reply submissions/bugagashinka/friends-app/index.html]: Done

File: submissions/bugagashinka/friends-app/index.html [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          <div class="order-name">
+            <h3>Name:</h3>
+            <label class="container"
+              >a-Z<input
+                type="radio"
+                name="order"
+                value="name-asc"
+                class="checked"
+              />
+              <span class="checkmark"></span>
```
[the remark this answers is not in the document]
Comment (bugagashinka) [reply submissions/bugagashinka/friends-app/index.html]: Thanks for your time on review, and for your suggestions for improvement.
I know the key difference between this types of component, but
I did not realize the idea that I planned, so we have this behavior.
Done.

File: submissions/bugagashinka/friends-app/js/components/DataModel.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+      isOrdered.byAge = false;
+      isOrdered.byName = true;
+    }*/
+/////
+/*isOrdered.asc = asc;
+    searchResult.sort(comparator);
+    console.log(searchResult);
+
+    prevFilterValue.order = `${prop}-${asc}`;
+    return searchResult;*/
```
[the remark this answers is not in the document]
Comment (bugagashinka) [reply submissions/bugagashinka/friends-app/js/components/DataModel.js]: Sorry for this, but whole this file 'DataModel.js' is outdated

File: submissions/bugagashinka/friends-app/js/components/FiltersContainer.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+    filter(filterCondition);
+    const orderCond = filterCondition.order.split('-');
+    order(orderCond[0], orderCond[1] == ORDER_ASC);
+    filterResultChange(filterResult);
+  };
+
+  searchField.addEventListener('keyup', event => {
+    if (event.code.indexOf('Key') != 0 && event.keyCode != BACK_SPACE_CODE)
+      return;
+    filterCondition.name = `${event.target.value}`;
```
[the remark this answers is not in the document]
Comment (bugagashinka) [reply submissions/bugagashinka/friends-app/js/components/FiltersContainer.js]: Done

**Pull request merged**